jerry-core/ecma/operations/ecma-promise-object.h

```diff
@@ -58,6 +58,23 @@ typedef struct
   ecma_collection_header_t *reject_reactions; /**< list of PromiseRejectReactions */
 } ecma_promise_object_t;
 
+/**
+ * Use symbolic constant to represent the internal property name of
+ * promise related structures.
+ */
+typedef enum
+{
+  ECMA_PROMISE_PROPERTY_PROMISE, /**< [[Promise]] property */
+  ECMA_PROMISE_PROPERTY_RESOLVE, /**< [[Resolve]] property */
+  ECMA_PROMISE_PROPERTY_REJECT, /**< [[Reject]] property */
+  ECMA_PROMISE_PROPERTY_CAPABILITY, /**< [[Capability]] property */
+  ECMA_PROMISE_PROPERTY_HANDLER, /**< [[Handler]] property */
+  ECMA_PROMISE_PROPERTY_ALREADY_CALLED, /**< [[AlreadyCalled]] property */
+  ECMA_PROMISE_PROPERTY_INDEX, /**< [[Index]] property */
+  ECMA_PROMISE_PROPERTY_VALUE, /**< [[Values]] property */
+  ECMA_PROMISE_PROPERTY_REMAINING_ELEMENT /**< [[RemainingElement]] property */
```

Comment:
Sorry @zherczeg , I didn't explain well. 

First of all, it is not a big perf issue, since the "prop name sharing" only appears in some routines.
And to this problem, there always be a tradeoff between perf and readability/maintainability. I dont think `str_capability = str_promise;` is a good idea, people may be confused: why str_capability can be assign with str_promise, why  str_resolve cannot? 

Since it will not hurt the perf much, I think it is best to keep the current code.